### ISSUE TITLE
Forward oidc-agent to docker container

### ISSUE THREAD

**Issue by ag14774** [indigo-dc/oidc-agent#605]
I am trying to forward the `oidc-agent` running on my computer to a docker container using the following command:
```
docker run -it -v /tmp/oidc-Ur9cew/oidc-agent.3465064:/oidc_sock --env OIDC_SOCK=/oidc_sock myalpine_image bash
```

However, when I run `oidc-token myoidc`, I get a GUI popup to type the encryption password (which is empty) to unlock. I directly press OK without typing anything and then I get:
```
root@f2303895f9ad:~# oidc-token myoidc
Error: account not loaded
```

Any idea as to what else I need to do to get this working? Is there anything else I need to forward?

**Comment by ag14774** [comment]
I also made the configs under .config/oidc-agent available to the container and that seems to work. However, I need to call oidc-add to load the account even though it is loaded on the host agent

**Comment by zachmann** [comment]
We recommend do use a non-empty encryption password - or use gpg-based encryption.

In principle it should be enough to forward the socket to the container. I also verified this using debian. I can obtain an AT from the container for an already loaded config, and also for a not yet loaded config using `oidc-token` and the gui prompting on the host. 
(Using `oidc-add` from the container would require to also mount the `.config/oidc-agent` dir).

Are you sure your `$OIDC_SOCK` points to the mounted host socket. Your comment sounds like it might not and you are using a agent running in the container.

There might be the case that oidc-token cannot auto-load with an empty encryption password, but if the config was loaded with `oidc-add` from the host, it should work anyway.

**Comment by ag14774** [comment]
I have confirmed that indeed it works if there is a non-empty password.

It does not work for the case of an empty password though. It gives me the message `Error: account not loaded`.

> Are you sure your $OIDC_SOCK points to the mounted host socket. Your comment sounds like it might not and you are using a agent running in the container

I'm sure because I have only copied `oidc-token` and `oidc-add` (plus required shared libraries) to the container. There is not `oidc-agent` binary

**Comment by zachmann** [comment]
It does not surprise me that the empty password does not work with the gui prompting. 

But if the account is already loaded and you can obtain a token from the host you should also be able to obtain it from the container.

**Comment by ag14774** [comment]
> But if the account is already loaded and you can obtain a token from the host you should also be able to obtain it from the container.

yeah it seem it does not. It still prompts me to type a password in the GUI. I can share a Dockerfile later to reproduce.

In any case, it seems having a password is the way to go :)

**Comment by zachmann** [comment]
I tried myself, and could confirm my expected behavior, if not loaded, the gui prompting indeed fails with the empty password, but when loaded outside the container, I can obtain one in the container.

But as you said indeed we recommend to use a password anyway.
